### ISSUE TITLE
Lack of examples

### ISSUE THREAD

**Issue by anonrig** [react-toolbox/react-toolbox#961]
Dear React-toolbox maintainers,

Even though I'm new to React, I have used similar frameworks in the past. I'm in the point of selecting the *best* material ui components for react, and I've come to a position that both libraries including react-box and material-ui lacks the examples of forms.

All I wanted to see is a basic login form with flex grid, and I couldn't find it on both libraries.

Did I miss it?

I've seen that both libraries are component based and doesn't give a general outline of a CSS framework. 

For example, Angular Material library includes community based examples on such functionality. 

http://codepen.io/ThomasBurleson/pen/aOXeXN

I would love to hear your feedback on this.

**Comment by borela** [comment]
The library itself isn't hard, the main thing to remember is how to bind the state using the `value` and `onChange` properties (which is what the [website](http://react-toolbox.com/#/components/input) focuses):

```Javascript
<Input
  type='text' 
  label='Name' 
  name='name' 
  value={this.state.name} 
  onChange={this.handleChange.bind(this, 'name')}  
  maxLength={16 }
/>
```

After that, the login logic is just normal react + javascript.... check this [tutorial](https://www.youtube.com/watch?v=tIajENrOJ0o). There's an [example](https://github.com/react-toolbox/react-toolbox-example) on how to load the library.

As for Material UI vs React Toolbox, I tried them both and what made me stop using Material UI is **inline CSS**, it makes the generated HTML to big and hard to debug/navigate, imagine a complex page with this:

![material](https://cloud.githubusercontent.com/assets/11317458/20513857/d8c4a1cc-b06f-11e6-8048-eb8f09398ad3.png)

**Comment by anonrig** [comment]
@borela Thank you for your reply. I believe I've been misunderstood on this topic. Let me rephrase it:

All I wanted to see is a ***basic login form with flex grid***, and I couldn't find it on both libraries.

The lack of examples of using component based frameworks with ***grids*** is frustrating. Not the component itself.

**Comment by borela** [comment]
Do you want to load an external grid framework?

**Comment by borela** [comment]
I think I understood what problem you are having, React Toolbox uses CSS Modules while most of these grid frameworks just use normal CSS. To prevent webpack from compiling them as CSS modules you need a rule to load CSS modules and another to load common CSS.

An idea is to have all your global CSS into a directory e.g. global-style, then you add a rule for it, this is a webpack 2 config:
```Javascript
let path = require('path');

let jsRule = {
  test: /\.jsx?$/,
  exclude: /node_modules/,
  use: [{
    loader: 'babel-loader',
    options: {
      plugins: [
        'transform-decorators-legacy',
        'transform-class-properties',
        'transform-export-extensions',
        'transform-object-rest-spread'
      ],
      presets: [
        ['es2015', {modules:false}],
        ['es2016'],
        ['es2017'],
        ['react']
      ]
    }
  }]
};

let globalCSSRule = {
  test: /\.css$/,
  exclude: /global-style/,
  use: [{
    loader: 'style-loader'
  },{
    loader: 'css-loader',
    options: {
      localIdentName: '[path][name]-[local]',
      sourceMap: true
    }
  },{
    loader: 'postcss-loader',
    options: {
      sourceMap: true
    }
  },{
    loader: 'sass-loader',
    options: {
      sourceMap: true,
      includePaths: [
        path.join(__dirname, 'src')
      ]
    }
  }]
};

let sassModuleRule = {
  test: /\.s[ac]ss$/,
  exclude: /global-style/,
  use: [{
    loader: 'style-loader'
  },{
    loader: 'css-loader',
    options: {
      localIdentName: '[path][name]-[local]',
      modules: true,
      sourceMap: true
    }
  },{
    loader: 'postcss-loader',
    options: {
      sourceMap: true
    }
  },{
    loader: 'sass-loader',
    options: {
      sourceMap: true,
      includePaths: [
        path.join(__dirname, 'src')
      ]
    }
  }]
};

module.exports = {
  resolve: {
    modules: [
      path.join(__dirname, 'src'),
      path.join(__dirname, 'node_modules')
    ],
    extensions: ['.js', '.scss']
  },
  entry: [
    'babel-polyfill',
    'react-toolbox/lib/commons',
    'app'
  ],
  output: {
    path: path.join(__dirname, 'build'),
    filename: 'script.js?[hash:8]'
  },
  devtool: 'source-map',
  devServer: {
    contentBase: 'build',
    historyApiFallback: true
  },
  module: {
    rules: [
      jsRule,
      globalCSSRule,
      sassModuleRule
    ]
  },
};
```

**Comment by anonrig** [comment]
Please take a look at: http://flexboxgrid.com/

![oops](https://media.giphy.com/media/11OJV2zUZRVL3y/giphy.gif)

**Comment by javivelasco** [comment]
Thanks for your responses @borela. I'm sorry but we can only accept examples from the community around react-toolbox. The main problem with this @anonrig is optimization of *time*. React Toolbox is a project that generates no benefits, is no backed and with a small community. There is a lot of stuff to do so personally I can't dedicate time to build full and complex examples with the library.

I think that if components are designed in an idiomatic way and well isolated, there shouldn't be any problems integrating with flexbox grids and anything you wan to wrap them in. That's the main purpose and right now the biggest problem is the amount of configuration you need to use css modules. The objective now is to reduce this amount of work making the library easier to adopt and we should put the focus on those topics while they are not solved instead of writing examples.

The idea is that you should be able to use react toolbox with other libraries and if the components are not behaving consistently or if they are breaking stuff out of the scope, then we are doing something wrong. Meanwhile, being a beginner to React, the best you can do is to try out tutorials to get used to the *react-way-of-thinking*.

That said, we are very open to PRs and projects serving as examples of react-toolbox!!
Thanks for your feedback :)

**Comment by coljung** [comment]
Yes, I was going to suggest: http://roylee0704.github.io/react-flexbox-grid/

**Comment by javivelasco** [comment]
PS: The documentation itself is an example

**Comment by colin-umn** [comment]
@javivelasco - I am trying to use the docs app as an example in the css-next branch.  Currently, it breaks on a commons.css dependency in app/index.js.  I suppose it might break in other places.  Is the docs app in css-next generally broken for now and not usable as a working example?

**Comment by javivelasco** [comment]
Hi @colin-umn ! Sorry, the docs are not updated for the css-branch so don't trust them too much. For now I can say there are no commons anymore. Each component has it's own autosufficient CSS and totally scoped and it shouldn't mess with the rest of your app. You can safely remove commons.css import :)
